src/meep.hpp

```diff
@@ -1490,30 +1508,41 @@ class fields {
 			double freq_min, double freq_max, int Nfreq);
 
   /********************************************************/
-  /* "flux operations" include things like                */
-  /*   (1) exporting dft_flux fields to HDF5 files        */
-  /*   (2) computing eigenmode decomposition coefficients */
-  /*       of dft_flux fields                             */
-  /* these are calculations that involve similar loops    */
-  /* over chunks, etc, so we consolidate them into a      */
-  /* single omnibus routine (do_flux_operation) with      */
-  /* multiple entry points for particular calculations.   */
+  /* process_dft_component is an intermediate-level       */
+  /* routine that serves as a common back end for several */
+  /* operations involving DFT fields (specifically,       */
+  /* writing DFT fields to HDF5 files and evaluating      */
+  /* overlap integrals between flux and mode fields.)     */
   /********************************************************/
-  void do_flux_operation(dft_flux flux, int num_freq, const volume where,
-                         const char *HDF5FileName,
-                         void *mode1_data=0, void *mode2_data=0,
-                         std::complex<double> *integrals=0);
-  void output_flux_fields(dft_flux flux, const volume where,
-                          const char *HDF5FileName);
+  std::complex<double> process_dft_component(dft_chunk **chunklists, 
+                                             int num_chunklists,
+                                             int num_freq, component c,
+                                             const char *HDF5FileName,
+                                             void *mode1_data=0, 
+                                             void *mode2_data=0,
+                                             component c_conjugate=Ex,
+                                             const volume *where=0);
+  
+  // output DFT fields to HDF5 file
+  void output_dft_components(dft_chunk **chunklists, int num_chunklists,
+                             const char *HDF5FileName, const volume *where=0);
+
+  void output_dft_flux(dft_flux flux, const char *HDF5FileName, const volume *where=0);
+  void output_dft_force(dft_force force, const char *HDF5FileName, const volume *where=0);
+  void output_dft_near2far(dft_near2far n2f, const char *HDF5FileName, const volume *where=0);
```

Comment:
Just rename these to `output_dft`, overloaded on the first argument.

Add a new `dft_fields` object that just takes the frequencies and a list of field components (`Ex`, `Hz`, etcetera) to Fourier transform in some volume, and support these in `output_dft`.